traffic_ops/traffic_ops_golang/user/user.go

```diff
@@ -0,0 +1,445 @@
+	user.LocalPassword = nil
+	user.ConfirmLocalPassword = nil
+
+	if rowsAffected != 1 {
+		if rowsAffected < 1 {
+			return fmt.Errorf("no user found with this id"), nil, http.StatusNotFound
+		}
+		return nil, fmt.Errorf("this update affected too many rows: %d", rowsAffected), http.StatusInternalServerError
+	}
+
+	return nil, nil, http.StatusOK
+}
+
+func (user *TOUser) Delete() (error, error, int) {
+	return api.GenericDelete(user)
+}
+
+func (u *TOUser) IsTenantAuthorized(user *auth.CurrentUser) (bool, error) {
+
+	// Delete: only id is given
+	// Create: only tenant id
+	// Update: id and tenant id
+	//	id is associated with old tenant id
+	//	we need to also check new tenant id
+
+	tx := u.ReqInfo.Tx.Tx
+
+	if u.ID != nil { // old tenant id (only on update or delete)
+
+		var tenantID int
+		if err := tx.QueryRow(`SELECT tenant_id from tm_user WHERE id = $1`, *u.ID).Scan(&tenantID); err != nil {
+			if err != sql.ErrNoRows {
+				return false, err
+			}
+
+			// At this point, tenancy isn't technically 'true', but I can't return a resource not found error here.
+			// Letting it continue will let it run into a 404 when it tries to update.
+			return true, nil
+		}
+
+		//log.Debugf("%d with tenancy %d trying to access %d with tenancy %d", user.ID, user.TenantID, *u.ID, tenantID)
+		authorized, err := tenant.IsResourceAuthorizedToUserTx(tenantID, user, tx)
+		if err != nil {
+			return false, err
+		}
+		if !authorized {
+			return false, nil
+
+		}
+	}
+
+	if u.TenantID != nil { // new tenant id (only on create or udpate)
+
+		//log.Debugf("%d with tenancy %d trying to access %d", user.ID, user.TenantID, *u.TenantID)
+		authorized, err := tenant.IsResourceAuthorizedToUserTx(*u.TenantID, user, tx)
+		if err != nil {
+			return false, err
+		}
+		if !authorized {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func (user *TOUser) SelectQuery() string {
+	return `
+	SELECT
+	u.id,
+	u.username as username,
+	u.public_ssh_key,
+	u.role,
+	r.name as rolename,
+	u.company,
+	u.email,
+	u.full_name,
+	u.new_user,
+	u.address_line1,
+	u.address_line2,
+	u.city,
+	u.state_or_province,
+	u.phone_number,
+	u.postal_code,
+	u.country,
+	u.registration_sent,
+	u.tenant_id,
+	t.name as tenant,
+	u.last_updated
+	FROM tm_user u
+	LEFT JOIN tenant t ON u.tenant_id = t.id
+	LEFT JOIN role r ON u.role = r.id`
+}
+
+func (user *TOUser) UpdateQuery() string {
+	return `
+	UPDATE tm_user u SET
+	username=:username,
+	public_ssh_key=:public_ssh_key,
+	role=:role,
+	company=:company,
+	email=:email,
+	full_name=:full_name,
+	new_user=COALESCE(:new_user, FALSE),
+	address_line1=:address_line1,
+	address_line2=:address_line2,
+	city=:city,
+	state_or_province=:state_or_province,
+	phone_number=:phone_number,
+	postal_code=:postal_code,
+	country=:country,
+	registration_sent=:registration_sent,
+	tenant_id=:tenant_id,
+	local_passwd=COALESCE(:local_passwd, local_passwd),
+	confirm_local_passwd=COALESCE(:confirm_local_passwd, confirm_local_passwd)
```

Comment:
you've removed this column from the db,  right?   won't this break?
As discussed in the mailing list,  confirmLocalPasswd needs to be removed from **all** of TO.  It should be handled only from the front-end in TP.

Reply:
At first glance, it appears not to have broken due to COALESCE, but I'm not sure. In any case, I'll remove it.

Reply:
Confirmed it doesn't break. Before I remove it though I'd like to confirm that we actually want to remove it from the API, despite it always being null.

Reply:
@dangogh

Reply:
fair enough -- we can clean up places where that column is referenced later on